ui/packages/catalog-ui-search/src/main/webapp/js/cql.js

```diff
@@ -23,7 +23,7 @@ define(['moment'], function(moment) {
     temporalClass = 'Temporal',
     timePatter = /([0-9]{4})(-([0-9]{2})(-([0-9]{2})(T([0-9]{2}):([0-9]{2})(:([0-9]{2})(\.([0-9]+))?)?(Z|(([-+])([0-9]{2}):([0-9]{2})))?)?)?)?/i,
     patterns = {
-      PROPERTY: /^([_a-zA-Z]\w*|"[^"]+")/,
+      PROPERTY: /^([_a-zA-Z]\w*|"[^"]+"|'[^']+')/,
```

Comment:
❗️ This is not CQL or ECQL compliant.  Compliant parsers might fail to parse these changes.  We should add tests to the back end to verify this will work and not regress.  If it works then we should also add a comment that this is not standard.

https://github.com/geotools/geotools/blame/20.0/modules/library/cql/ECQL.md#L398-L399

Reply:
This change is only used for parsing CQL on the frontend, so the flexibility is to handle non-standard ECQL on the frontend, no changes to what is sent to the backend (logic in cql.write hasn't been changed). I added a comment to the JS that it allows non-standard single quote property names.

Reply:
This change is to support passing the list.cql in listTemplates to the backend. The backend should receive an escaped double quote, but because of some bugs in Felix fileinstall and Felix utils those get improperly parsed and not passed correctly causing the cql.js parser to fail. Fixing those dependencies is going to be more involved as DDF and Karaf aren't on the latest version and replacing the existing version within Karaf & DDF is going to lead to some hacking of the baseline, which I think is a little worse than making this change.